### PR TITLE
request to merge Game branch with development branch

### PR DESCRIPTION
**Pull request by mohamednaser**
we finished below tasks : 
1- form to get players names 
2- design the game 
3- build winner logic 
4- build helper buttons to finish the game or restart it

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        <div class="form-group">
+          <label for="playerOLabel">Second Player Name : </label>
+          <input type="text" class="form-control" id="playerO" />
+        </div>
+        <div class="col text-center">
+          <button
+            type="button"
+            class="btn btn-primary text-center"
+            id="start-game"
+            onclick="game.start()"
```
Comment (azdrenymeri) [inline index.html]: It is not a good practice to mix javascript with HTML.
You have already added the id, so on your js file
```javascript
document.querySelector('#start-game').addEventListener('click', () => {
game.start();
});
```

File: index.html [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                  <td id="playerOName">second name</td>
+                  <td id="oWins">0</td>
+                </tr>
+              </tbody>
+            </table>
+            <div class="game-buttons">
+              <button
+                class="btn btn-danger"
+                id="finish-game"
+                onclick="game.restart();"
```
Comment (azdrenymeri) [inline index.html]: Same as line 36.

File: index.html [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+              >
+                Finish Game
+              </button>
+              <button
+                class="btn btn-primary"
+                id="play-again-game"
+                onclick="board.reset()"
```
Comment (azdrenymeri) [inline index.html]: Same as line 36

File: js/custom.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
+// Global Tic-tak-toe Variables
```
Comment (azdrenymeri) [inline js/custom.js]: Kindly remove all the comments on the file.

File: js/custom.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    boardArray[selectedRowColumeArray[0]][
+      selectedRowColumeArray[1]
+    ] = currentPlayer;
+    if (game.checkWin()) {
+      // show winner and finish this round
+      alert(currentPlayer + "win");
+      game.setWinner(currentPlayer);
+    } else {
+      game.switchPlayer();
+      console.log(selectedCell.id, "selectedCell details ", boardArray);
```
Comment (azdrenymeri) [inline js/custom.js]: Remove the `console.log()` statements

File: js/custom.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+      for (let y = 0; y <= 2; y++) {
+        let htmlCeil = `<div class='board-cell cursor-pointer' onclick="game.move(this)" id="${x},${y}"> ${boardArray[x][y]}</div>`;
+
+        boardArrayDiv.innerHTML += htmlCeil;
+      }
+    }
+  },
+
+  reset: () => {
+    console.log("players data ", playerX, playerO);
```
Comment (azdrenymeri) [inline js/custom.js]: Remove the `console.log()` statement.

**Review by azdrenymeri — CHANGES_REQUESTED** [review]
### Hello @mohamednaser  :wave: 
- [x]  `.git-ignore` file should be set to `.gitignore`(without the dash) so `Git` can recognize it.
Because of this typo all your project's `node_modules` are published on the remote branch so make sure to fix this.
- [x] Also stickler is not working(I'm guessing after fixing the `.gitignore` issue it will work), kindly check it out.
- [x] I can start the game without providing the player's name, let's add some validation on the form.
- [x] I see you have used Unobtrusive JavaScript on buttons in `index.html` kindly separate javascript from HTML.
#### Implement the changes and add it for another review

**Review by mcrd25 — APPROVED** [review]
# Changes Approved 💪 🎉 🚀 

Hey @mohamednaser 👋 

Good job implementing changes requested from the previous review!
There is no need for further review now! 
Congratulations! ㊗️ 

Also, your game UI is very minimal and neat, I like it! 😄 

🚀 

***If you have any questions, now or in the future, do not hesitate to contact me on social media, email or on slack and I will be glad to chat with you.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

**Pull request merged**